File: first-trade.js
var request = require('request'),
  keys = require("./keys.js");

var   apiKey = keys.apiKey,
      venue = "FOOEX",
      stock = "HOGE",
      base_url = "https://api.stockfighter.io/ob/api",
      account = keys.account;

var order = {
  "account" : account,
  "venue" : venue,
  "symbol" : stock,
  "price" : 95,
  "qty" : 10,
  "direction" : "buy",
  "orderType" : "limit"
};

var headers = {
  "X-Starfighter-Authorization" : apiKey
};

var options = {
  method : "POST",
  url: base_url + "/venues/" + order.venue + "/stocks/" + stock + "/orders",
  headers: headers
};

var theRequest = function(options,callback){

 var req = request.get(options);

 req.on('response', function(res){
   var chunks = [];

   res.on('data', function(chunk){
     chunks.push(chunk);
   });

    res.on('end', function(){
      var buffer = Buffer.concat(chunks);
      var encoding = res.headers['content-encoding'];

      if (encoding == 'gzip'){
        zlib.gunzip(buffer, function(err, decoded){
          callback(err, decoded && decoded.toString());
        });
      } else if (encoding == 'deflate'){
        zlib.inflate(buffer, function(err, decoded) {
          callback(err, decoded && decoded.toString());
        });
      } else {
        callback(null, buffer.toString());
      }
    });
  });

  req.on('error', function(err){
    callback(err);
  });

};

var theCall = function(){
  theRequest(options, function(err, data) {
      if (err) console.log('error', err);
      else{
        console.log('the response:');
        console.log(data);
      }
    }
  );
};

theCall();

File: first-trade.rb
#this ruby example came from stockfighter
require 'rubygems'
require 'httparty'
require 'json'

apikey = "123456123456123456123456123456123456123456"
venue = "FOOEX"   # Replace this with your real value.
stock = "HOGE"  #Fun fact: Japanese programmers often use "hogehoge" where Americans use "foobar."  You should probably replace this with your real value.
base_url = "https://api.stockfighter.io/ob/api"

account = "HB61251714"  # Printed in bold in the level instructions. Replace with your real value.

# Set up the order

order = {
  "account" => account,
  "venue" => venue,
  "symbol" => stock,
  "price" => 25000,  #$250.00 -- probably ludicrously high
  "qty" => 100,
  "direction" => "buy",
  "orderType" => "limit"  # See the order docs for what a limit order is
}


response = HTTParty.post("#{base_url}/venues/#{venue}/stocks/#{stock}/orders",
                         :body => JSON.dump(order),
                         :headers => {"X-Starfighter-Authorization" => apikey}
                         )

#Now we analyze the order response

puts response.body

### Here is what the response looked like.

# {
#   "ok": true,
#   "symbol": "HOGE",
#   "venue": "FOOEX",
#   "direction": "buy",
#   "originalQty": 100,
#   "qty": 0,
#   "price": 25000,
#   "orderType": "limit",
#   "id": 6408,
#   "account": "HB61251714",
#   "ts": "2015-08-18T04:00:08.340298024+09:00",
#   "fills": [
#     {
#       "price": 5960,
#       "qty": 100,
#       "ts": "2015-08-18T04:00:08.340299592+09:00"
#     }
#   ],
#   "totalFilled": 100,
#   "open": false
# }

# As we can see, I got 100 fills of the 100 shares I ordered.  Whee!
# This order is now closed (open: false).
